src/server/pkg/collection/collection.go

```diff
@@ -376,6 +382,21 @@ func (c *readonlyCollection) GetByIndex(index Index, val interface{}) (Iterator,
 	}, nil
 }
 
+func (c *readonlyCollection) GetBlock(key string, val proto.Unmarshaler) error {
```

Comment:
Definitely, although that's not quite what `GetBlock` does, it just blocks until there's a value. So if there's already a value it's behavior is identical to `Get` it's just that if the value doesn't exist yet `GetBlock` will block until it does. I'll add a comment to this effect, do you think there's maybe a name that better suggests that behavior. I considered `GetSafe` because it's safe to call concurrently with Create.